tests/test_utilities.py

```diff
@@ -156,6 +156,159 @@ def test_check_all_contiguous():
     print 'time for %s = %.2f'%(funcname(),t2-t1)
 
+def test_interleaveImages():
+    import time
+    t1 = time.time()
+
+    # 1a) With galsim Gaussian
+    g = galsim.Gaussian(sigma=3.7,flux=1000.)
+    gal = galsim.Convolve([g,galsim.Pixel(1.0)])
+    im_list = []
+    offset_list = []
+    n = 2
+    for j in xrange(n):
+        for i in xrange(n):
+            im = galsim.Image(16*n,16*n)
+            offset = galsim.PositionD(-(i+0.5)/n+0.5,-(j+0.5)/n+0.5)
+            offset_list.append(offset)
+            gal.drawImage(image=im,scale=1.0,method='no_pixel',offset=offset)
+            #gal.drawImage(image=im,scale=1.0,method='no_pixel',offset=galsim.PositionD(1.*i/n,1.*j/n))
+            im_list.append(im)
+
+    # Input to N as an int
+    img = galsim.utilities.interleaveImages(im_list,n,offsets=offset_list)
+    im = galsim.Image(16*n*n,16*n*n)
+    g = galsim.Gaussian(sigma=3.7,flux=1000.*n*n)
+    gal = galsim.Convolve([g,galsim.Pixel(1.0)])
+    gal.drawImage(image=im,method='no_pixel',offset=galsim.PositionD(0.0,0.0),scale=1.0/n)
+    np.testing.assert_array_equal(img.array,im.array,\
+        err_msg="Interleaved Gaussian images do not match")
+    assert im.scale == img.scale
+
+    # 1b) With im_list and offsets permuted
+    offset_list = []
+    # An elegant way of generating the default offsets
+    DX = np.arange(0.0,-1.0,-1.0/n)
+    DX -= DX.mean()
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+            offset = galsim.PositionD(dx,dy)
+            offset_list.append(offset)
+
+    np.random.seed(42) # for generating the same random permutation everytime
+    rand_idx = np.random.permutation(len(offset_list))
+    im_list_randperm = [im_list[idx] for idx in rand_idx]
+    offset_list_randperm = [offset_list[idx] for idx in rand_idx]
+    # Input to N as a tuple
+    img_randperm = galsim.utilities.interleaveImages(im_list_randperm,(n,n),offsets=offset_list_randperm)
+
+    np.testing.assert_array_equal(img_randperm.array,img.array,\
+        err_msg="Interleaved images do not match when 'offsets' is supplied")
+    assert img_randperm.scale == img.scale
+
+    # 1c) Catching errors if offsets
+    offset_list = []
+    im_list = []
+    n = 5
+    DX = np.array([-0.67,-0.33,0.,0.33,0.67])
+    DY = DX
+    for dy in DY:
+        for dx in DX:
+           offset = galsim.PositionD(dx,dy)
+           offset_list.append(offset)
+           im = galsim.Image(16,16)
+           gal.drawImage(image=im,offset=offset,scale=1.0,method='no_pixel')
+           im_list.append(im)
+
+    VE = ValueError('No error')
+    try:
+        N = (n,n)
+        galsim.utilities.interleaveImages(im_list,N=N,offsets=offset_list,catch_offset_errors=True)
+    except ValueError as VE:
+        pass
+    message =  "'offsets' must be a list of galsim.PositionD instances with x values"\
+                          +" spaced by 1/{0} and y values by 1/{1} around 0 for N = ".format(n,n)+str(N)
+    assert VE.message == message
```

Comment:
We normally use the `assert_raises` function for this.  cf. test_wfirst.py (among many others).  You do need to wrap it in a try/except block to catch an ImportError on systems that don't have nose installed.  But it's a bit cleaner than do this manually.